### PR TITLE
Load pac bio product id

### PR DESCRIPTION
**Pull request by mksanger**
Add a python script, dependent on the https://github.com/wtsi-npg/npg_id_generation repo, to generate product ids - I don't know whether this is the best place for the script, but it simplifies paths.  I'm also not sure about how best to declare cross language dependencies.

Generate and load product ids to the pac_bio_run_well_metrics and pac_bio_product_metrics table.

### CODE REVIEW

File: bin/npg_pacbio_runs2mlwarehouse [hunk 1]
```diff
@@ -19,6 +19,7 @@ our $VERSION = '0';
 Readonly::Scalar my $DEFAULT_INTERVAL_DAYS   => 100_000;
 Readonly::Scalar my $DEFAULT_OLDER_THAN_DAYS => 0;
 Readonly::Scalar my $DEFAULT_HOSTNAME => q[pacbio02.dnapipelines.sanger.ac.uk];
+Readonly::Scalar my $DEFAULT_ID_SCRIPT => q[/software/npg/current/bin/generate_pac_bio_id];
```
Comment (mgcam) [inline bin/npg_pacbio_runs2mlwarehouse:23]: We should never hardcode the location of the dependencies

File: MANIFEST [hunk 2]
```diff
@@ -1,5 +1,6 @@
 .dir-locals
 .github/workflows/testing_and_building_repo.yml
+bin/generate_pac_bio_id
```
Comment (mgcam) [inline MANIFEST:3]: The script is not part of this repo

File: lib/npg_warehouse/loader/pacbio/run.pm [hunk 3]
```diff
@@ -51,6 +51,12 @@ has 'run_uuid' =>
    required      => 1,
    documentation => 'The PacBio run unique identifier',);
 
+has 'id_script' =>
+  (isa           => 'Str',
+   is            => 'ro',
+   required      => 1,
+   documentation => 'A path to the script used to generate product ids',);
+
```
Comment (mgcam) [inline lib/npg_warehouse/loader/pacbio/run.pm:59]: The convention is to rely on executable scripts being on the PATH, no need to pass around a path to the script.

File: lib/npg_warehouse/loader/pacbio/run.pm [hunk 4]
```diff
@@ -174,6 +180,15 @@ sub _build_run_wells {
       my $tki = $self->_well_tracking_info($well->{'uniqueId'});
       my $run = $self->_run;
 
+      open my $id_product_script, q[-|],
+        join q[ ], 'python', $self->id_script,
```
Comment (mgcam) [inline lib/npg_warehouse/loader/pacbio/run.pm:184]: Reading from the output of the process is slightly different from reading from a file handle. We have to deal with a case when the process errors.

File: lib/npg_warehouse/loader/pacbio/run.pm [hunk 5]
```diff
@@ -174,6 +180,15 @@ sub _build_run_wells {
...
+          $run->{'pac_bio_run_name'}, $well->{'well'}
+        or $self->logconfess ('Cannot generate id_product');
+      my $id_product = <$id_product_script>;
+      $id_product =~ s/\s//xms;
```
Comment (mgcam) [inline lib/npg_warehouse/loader/pacbio/run.pm:188]: Here we assume that we get a value. However, we actually have to check that. We also might want to check that we get a string of the expected length.

**Comment by mksanger** [comment]
Failing due to lack of https://github.com/wtsi-npg/ml_warehouse/pull/195

File: Changes [hunk 6]
```diff
@@ -1,5 +1,7 @@
 LIST OF CHANGES
 
+ - add product id generation and loading to
```
Comment (mgcam) [inline Changes:3]: The sentence looks unfinished

**Comment by mgcam** [comment]
Pr tested locally. @mksanger loaded mixed data (one or multiple samples per well) to the dev database.

**Pull request merged**